### PR TITLE
Capture server-side metrics for Protobuf RPCs

### PR DESCRIPTION
**Pull request by zfrenette**
Fix and capture micrometer server-side metrics for Protobuf RPCs.

## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [X] There are no TODOs left in the code
- [X] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [X] Change is covered by automated tests
- [X] Public API has Javadoc

### CODE REVIEW

File: infrastructure/src/main/java/org/corfudb/infrastructure/RequestHandlerMethods.java [hunk 1]
```diff
@@ -130,10 +133,50 @@ private void registerMethod(@Nonnull final MethodHandles.Lookup caller,
...
+
+    /** Generate a conditional handler, which instruments the handler with metrics if configured
+     * as enabled and checks whether the server is shutdown and ready.
+     * @param type          The type the request message is being handled for.
+     * @param handler       The {@link RequestHandlerMethods.HandlerMethod} which handles the message.
+     * @return              A new {@link RequestHandlerMethods.HandlerMethod} which conditionally executes the
+     *                      handler based on preconditions (whether the server is shutdown/ready).
+     */
+    private HandlerMethod generateConditionalHandler(@NonNull final PayloadCase type,
+                                                     @NonNull final HandlerMethod handler) {
```
Comment (xcchang) [inline infrastructure/src/main/java/org/corfudb/infrastructure/RequestHandlerMethods.java:153]: nit: the javadoc's indent is broken here

**Review by xcchang — APPROVED** [review]
Looks great!

File: infrastructure/src/main/java/org/corfudb/infrastructure/RequestHandlerMethods.java (hunk 1 above)
Comment (zfrenette) [reply infrastructure/src/main/java/org/corfudb/infrastructure/RequestHandlerMethods.java]: Fixed.

**Review by PavelZaytsev — APPROVED** [review]
lgtm

**Pull request merged**